src/DGtal/base/CowPtr.h

```diff
@@ -88,14 +89,19 @@ namespace DGtal
     const T& operator*()    const throw()   {return *myPtr;}
     const T* operator->()   const throw()   {return myPtr.get();}
     const T* get()          const throw()   {return myPtr.get();}
-    
+
+    template < typename U = T, typename std::enable_if< ! std::is_const<U>::value >::type* = nullptr >
```

Comment:
Do you mind commenting a bit these lines ? (Like forces the compiler to choose the const version without copy whenever possible)